### PR TITLE
Make it so name trackers require approval to send spam notification to observer.

### PR DESCRIPTION
**Pull request by pellaea**
Adds name_trackers.approved column, defaulting to true for all existing name_trackers with non-empty note_template.

When new name_trackers are created with a note_template, it starts as unapproved, and sends a notification to the admins.  The admin notification has a link to approve the name_tracker.

All unapproved name_tracker notifications to the observer are ignored.

A message is displayed for the tracker letting them know that their request is pending approval.

They can check if it has been approved in "Your Interests".

-----

Manual test:

1. go to any random observation
2. go to "About xxx" (name page)
3. click on "Email Tracking" in upper right
4. check the "Send a research request to the observer? " checkbox to enable the spammy half of name tracking, then submit the form

You should see a green flash notice letting you know that your request to send notifications to observers is pending approval, along with a link to "Your Interests" where you can see all of your name_tracker notification requests.  On that page you should see your new name_tracker with the note "(Observer notification pending approval.) "

5. grab the name_tracker id from hovering over the "Change to Ignore" or "Destroy" links (not the "Show Name Tracker" link -- that one has the name id instead of the name_tracker id)
6. type in http://localhost:3000/name/approve_tracker/<id> (inserting the appropriate id from 5)
7. now go back to "Your Interests"

The "pending approval" business shouldn't be there any more.  Note that it does a flash notice "Name stalker approved" when an admin approves of a name_tracker.  If you are logged in as a user who is not an admin it will fail.  But you do not have to actually be in admin mode when you request the /name/approve_tracker action.

### CODE REVIEW

**Comment by pellaea** [comment]
How important is it to send a notification to the tracker when we approve their request?  I guess if we make it a very short, simple email, ASCII-only, it shouldn't be that much work...

**Comment by JoeCohen** [comment]
> How important is it to send a notification to the tracker when we approve their request? I guess if we make it a very short, simple email, ASCII-only, it shouldn't be that much work...

Maybe nice to have, but not critical. A reasonable alternative is:
- manually email the user about approval (or rejection). (If/when we reject, we should explain.)
- include a reminder to email the user when the admin is notified about the new request.

**Comment by JoeCohen** [comment]
Seems like I have forgotten everything.
I wanted to try a short manual test, but cannot login. What am I doing wrong?
>`rails db:migrate`
`rails lang:update`
`rails s -b 0.0.0.0`
Tried logging in with various combinations of my user name, email, login and "password" || my real password

**Comment by JoeCohen** [comment]
Never mind. I forgot to load the db. (At least I was right about forgetting everything.)

**Comment by JoeCohen** [comment]
Should `app/views/interests/index.html.erb` display something like "Pending Approval" if a Notification Request is not (yet) approved?

**Comment by pellaea** [comment]
Yes, it should display something about approval pending if the name_tracker
is not approved yet.

**Comment by pellaea** [comment]
Added instructions for manual test in comment.

Also, note the activity log-style interest-type buttons that are now shown on the "Your Interests" index.  If you don't have any interests, then it probably won't show any.  Try using the switch-users feature to my account (login=jason) and see what my interests index looks like for a nontrivial example.   Play around with those button to make sure they work right and that paginations continues to work in each case.

**Review by JoeCohen — APPROVED** [review]
Looks good!
Thanks for the manual test. I ran through it, including the additional instructions.

**Comment by pellaea** [comment]
Good idea, thanks, Joe.

File: app/controllers/name_controller.rb
Comment (pellaea) [reply app/controllers/name_controller.rb:1150]: Done, I'll check it in in a minute.

**Comment by JoeCohen** [comment]
Argh! All that work and there's still an ABC error. So close  <2, 20, 3> 20.32/20.
I will simply disable the cop for the offending method.

**Comment by JoeCohen** [comment]
This is another one I'd like to merge if there are no objections.

**Review by nimmolo — APPROVED** [review]
Approve!

**Comment by pellaea** [comment]
Yep, thank you, Joe!

**Pull request merged**
